browser/ui/sidebar/sidebar_controller.cc

```diff
@@ -103,9 +104,21 @@ void SidebarController::ActivateItemAt(std::optional<size_t> index,
   DCHECK_LT(index.value(), sidebar_model_->GetAllSidebarItems().size());
 
   const auto& item = sidebar_model_->GetAllSidebarItems()[*index];
+
+  if (IsWebPanelFeatureEnabled() && item.is_web_panel_type()) {
+    GetWebPanelController()->IsShowingWebPanel()
+        ? GetWebPanelController()->CloseWebPanel()
+        : GetWebPanelController()->OpenWebPanel(item);
+    return;
+  }
+
   // Only an item for panel can get activated.
   if (item.open_in_panel) {
-    sidebar_model_->SetActiveIndex(index);
+    // TODO(https://github.com/brave/brave-browser/issues/33533): WebPanel item
+    // type also should be activated.
+    if (!item.is_web_panel_type()) {
+      sidebar_model_->SetActiveIndex(index);
+    }
```

Comment:
Q. I'm confused why this was added.

Reply:
This web panel type item also should be in active state like other builtin panel(ex, playlist). but it's not ready yet. In this PR, clicking item just toggles web panel. Later we should call SetActiveIndex for web panel type.

Reply:
I see. Thanks :) One more question, I'm curious if all items in |open_in_panel| condition are not web panel types because it seems
* When the feature is enabled, returning from L112?
* When the feature is disabled, open_in_panel can't be true with web panel type?

Reply:
Oh, enabling feature -> add web panel -> disabling the feature can make this happen.